### PR TITLE
Updates for Julia 0.6

### PR DESCRIPTION
**Pull request by andreasnoack**
Mainly the deprecation of the old `Array` constructor. I'm also using `Ref` in `ccall`s whenever possible. Finally, I'm dropping Julia 0.3 support since some of these changes won't work there and I don't think it is relevant to support anymore.

### CODE REVIEW

**Comment by kbarbary** [comment]
Thanks for this PR! I definitely agree we can drop Julia 0.3 support.

Looks like `String(take!(io))` in runtests.jl is problematic on Julia 0.4 since it means `AbstractString(take!(io))` there. Maybe we need a custom Julia version check here?

Alternatively, we could also drop Julia 0.4 support.

**Comment by andreasnoack** [comment]
I had forgotten to `import Compat.String`. Should work on 0.4 now as well.

**Comment by kbarbary** [comment]
Great, thanks!

**Comment by andreasnoack** [comment]
Would you mind making a new release? If you don't already know attobot, you should take a look at https://github.com/attobot/attobot. It makes it much simpler.

File: test/runtests.jl [hunk 1]
```diff
@@ -136,7 +137,7 @@ f = FITS(fname, "w")
 # arre what we expect.
 io = IOBuffer()
 show(io, f)
-s = takebuf_string(io)
+s = @compat(String(take!(io)))
```
Comment (tkelman) [inline test/runtests.jl]: update Compat minimum version for this - 0.9.5

Comment (kbarbary) [inline test/runtests.jl]: Done.

**Comment by kbarbary** [comment]
JuliaLang/METADATA.jl#7653

Attobot is nice.

**Pull request merged**